### PR TITLE
Misc performance changes

### PR DESCRIPTION
**Pull request by cjolivier01**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [ ] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/imperative/imperative_utils.h [hunk 1]
```diff
@@ -453,7 +453,7 @@ inline void PushOperator(const OpStatePtr& state,
         << "for stateful operator " << op->name;
     CHECK(exec_type == ExecType::kSync || exec_type == ExecType::kAsync);
     Engine::Get()->PushSync(
-      [state, fcompute, inputs, outputs, requested, is_train, exec_type, mutate_idx, req](
+      [&state, fcompute, &inputs, &outputs, &requested, is_train, exec_type, &mutate_idx, &req](
```
Comment (piiswrong) [inline src/imperative/imperative_utils.h]: I don't think you can do this.

Comment (cjolivier01) [reply src/imperative/imperative_utils.h]: I was wondering why they were passed by value. Is Push() synchronous? This was my understanding.

Comment (cjolivier01) [reply src/imperative/imperative_utils.h]: That this lambda is executed before this Push() call returns? Now that I think about it I probably should have verified that but I just always thought this was the case.

Comment (cjolivier01) [reply src/imperative/imperative_utils.h]: Yeah, I guess it isn't synchronous like its comment description says. Removed.

Comment (piiswrong) [reply src/imperative/imperative_utils.h]: PushSync means the lambda is sync, but push itself is async

**Comment by cjolivier01** [comment]
the will be in the other PR, so closing this

**Pull request closed without merging**